src/Microsoft.OData.ModelBuilder/Property/PrimitivePropertyConfigurationExtensions.cs

```diff
@@ -46,6 +46,29 @@ public static PrimitivePropertyConfiguration AsTimeOfDay(this PrimitivePropertyC
                 throw Error.ArgumentNull("property");
             }
 
+            if (!TypeHelper.IsTimeSpan(property.RelatedClrType) && !TypeHelper.IsTimeOnly(property.RelatedClrType))
+            {
+                throw Error.Argument("property", SRResources.MustBeTimeSpanProperty, property.PropertyInfo.Name,
+                    property.DeclaringType.FullName);
+            }
+
+            property.TargetEdmTypeKind = EdmPrimitiveTypeKind.TimeOfDay;
```

Comment:
Was the `TimeOfDay` enum member going to change to `TimeOnly` or did we decide to save ourselves the trouble?